src/cmake/thirdparty/SetupADIOS2.cmake

```diff
@@ -10,21 +10,39 @@
 #
 ###############################################################################
 
-# first Check for ADIOS_DIR
-if(NOT ADIOS2_DIR)
-    MESSAGE(FATAL_ERROR "ADIOS2 support needs explicit ADIOS2_DIR")
+# Handle legacy usage of ADIOS2_DIR
+if (ADIOS2_DIR AND NOT ADIOS2_ROOT)
+  # If find_package(ADIOS2) has already been called this will fail
+  if (NOT EXISTS ${ADIOS2_DIR}/lib/cmake/adios2)
+    get_filename_component(tmp "${ADIOS2_DIR}" DIRECTORY)
+    get_filename_component(tmp "${tmp}" DIRECTORY)
+    get_filename_component(tmp "${tmp}" DIRECTORY)
+    if (EXISTS ${tmp}/lib/cmake/adios2)
+      set(ADIOS2_ROOT "${tmp}" CACHE PATH "")
+    else ()
+      message(FATAL_ERROR "Could not determine ADIOS2_ROOT from ADIOS2_DIR")
+    endif ()
+  else ()
+    set(ADIOS2_ROOT "${ADIOS2_DIR}" CACHE PATH "")
+  endif ()
+endif ()
+
+# Check for ADIOS_ROOT
+if(NOT ADIOS2_ROOT)
+    MESSAGE(FATAL_ERROR "ADIOS2 support needs explicit ADIOS2_ROOT")
 endif()
 
-MESSAGE(STATUS "Looking for ADIOS2 using ADIOS2_DIR = ${ADIOS2_DIR}")
+MESSAGE(STATUS "Looking for ADIOS2 using ADIOS2_ROOT = ${ADIOS2_ROOT}")
 
-set(ADIOS2_DIR_ORIG ${ADIOS2_DIR})
+set(ADIOS2_DIR_ORIG ${ADIOS2_ROOT})
 
 find_package(ADIOS2 REQUIRED
              NO_DEFAULT_PATH
-             PATHS ${ADIOS2_DIR}/lib/cmake/adios2)
+             PATHS ${ADIOS2_ROOT}/lib/cmake/adios2)
```

Comment:
So this line is more problematic than I originally thought. I am getting an issue with the installation of ADIOS2 not always being under the `lib/cmake/adios2` directory. Currently I am testing on crusher, and the install is under `lib64/cmake/adios2`.

Reply:
Yes, we had to deal with this style for Kokkos as well:

https://github.com/Alpine-DAV/ascent/blob/develop/src/cmake/thirdparty/SetupKokkos.cmake

Reply:
I believe just specifying the `PATH` as the `Package_ROOT` should be sufficient. CMake should search under that path following the standard installations locations even if `NO_DEFAULT_PATH` is used. I tested it this way on crusher and was successful. So unless you have multiple installations of adios2 in the same location, which you couldn't anyway, it will find it reguardless of the `lib` directory name.

https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure

Reply:
Sounds good, good to see things are evolving to be hopefully simpler over time :-)